x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/topn/TopNOperator.java

```diff
@@ -273,6 +273,14 @@ public String describe() {
     private final List<SortOrder> sortOrders;
 
     private Row spare;
+    private int spareValuesPreAllocSize = 0;
+    private int spareKeysPreAllocSize = 0;
+
+    /**
+     * Counts how often we needed to resize rows. Only works if assertions are enabled.
+     */
+    private int rowResizes;
```

Comment:
This is not about the memory usage, but I think this class is simpler without it. We can test the number of resizes with a custom circuit breaker in tests.